doc/glossary.rst

```diff
@@ -0,0 +1,1216 @@
+        * Some parallelism uses a multi-threading backend by default, some
+          a multi-processing backend.  It is possible to override the default
+          backend by using :func:`sklearn.externals.joblib.parallel.parallel_backend`.
+        * Whether parallel processing is helpful at improving runtime depends
+          on many factors, and it's usually a good idea to experiment rather
+          than assuming that increasing the number of jobs is always a good
+          thing.  *It can be highly detrimental to performance to run multiple
+          copies of some estimators or functions in parallel.*
+
+        Nested uses of ``n_jobs``-based parallelism with the same backend will
+        result in an exception.
+        So ``GridSearchCV(OneVsRestClassifier(SVC(), n_jobs=2), n_jobs=2)``
+        won't work.
+
+        When ``n_jobs`` is not 1, the estimator being parallelized must be
+        picklable.  This means, for instance, that lambdas cannot be used
+        as estimator parameters.
+
+    ``random_state``
+        Whenever randomization is part of a Scikit-learn algorithm, a
+        ``random_state`` parameter may be provided to control the random number
+        generator used.  Note that the mere presence of ``random_state`` doesn't
+        mean that randomization is always used, as it may be dependent on
+        another parameter, e.g. ``shuffle``, being set.
+
+        ``random_state``'s value may be:
+
+            None (default)
+                Use the global random state from :mod:`numpy.random`.
+
+            An integer
+                Use a new random number generator seeded by the given integer.
+                To make a randomized algorithm deterministic (i.e. running it
+                multiple times will produce the same result), an arbitrary
+                integer ``random_state`` can be used. However, it may be
+                worthwhile checking that your results are stable across a
+                number of different distinct random seeds.
+
+            A :class:`numpy.random.RandomState` instance
+                Use the provided random state, only affecting other users
+                of the same random state instance.
+
+        :func:`utils.check_random_state` is used internally to validate the
+        input ``random_state`` and return a :class:`~numpy.random.RandomState`
+        instance.
+
+    ``scoring``
+        Specifies the score function to be maximized (usually by :ref:`cross
+        validation <cross_validation>`), or -- in some cases -- multiple score
+        functions to be reported. The score function can be a string accepted
+        by :func:`metrics.get_scorer` or a callable :term:`scorer`, not to be
+        confused with an :term:`evaluation metric`, as the latter have a more
+        diverse API.  ``scoring`` may also be set to None, in which case the
+        estimator's ``score`` method is used.  See :ref:`scoring_parameter` in
```

Comment:
not sure if it's worth mentioning default of accuracy / R^2 here? Probably link to the explanation of the ``score`` method instead?